x/posts/keeper/handler.go

```diff
@@ -37,6 +37,20 @@ func NewHandler(keeper Keeper) sdk.Handler {
 	}
 }
 
+// IsCreatorBlockedBySomeTags checks if some of the post's tags have blocked the post's creator
+func IsCreatorBlockedBySomeTags(ctx sdk.Context, keeper Keeper, attachments types.Attachments, creator sdk.AccAddress) error {
+	for _, attachment := range attachments {
+		for _, tag := range attachment.Tags {
+			// check if the request's receiver has blocked the sender before
+			if keeper.IsUserBlocked(ctx, tag, creator) {
+				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+					fmt.Sprintf("The user with address %s has blocked you", tag))
+			}
+		}
+	}
+	return nil
```

Comment:
By taking a closer look at this code, I think that it could be further improved. 

Right now, what happens when this call is done is the following:

1. For each attachment, all tags are extracted. 
2. For each tag, we check if that tag is blocked. 

The thing is that to do (2) we: 

1. Extract all the users that have been blocked by the post creator
2. We iterate over that set 

This actually causes this method to access the store `m` times where `m` is the number of tags, causing the used gas to increment. 

Instead, what we could do is define a method such as 

```go
func (k Keeper) IsAnyUserBlocked(ctx sdk.Context, sdk.AccAddress blocker, ...sdk.AccAddress users) (bool, sdk.AccAddress) {
  blockedUsers := k.GetBlockedUsers(ctx, blocker)
  for _, user := range users {
    if blockedUsers.Contains(user) {
      return true, user
    }
  }
  return false, nil
}
```

This way we would have only one access to the store.  
(Note: The `Contains` method might be implemented inside the `common.go` file where other utils method lie).

This could then be used as follows:

```go
var tags []sdk.AccAddress
for _, attachment := range attachments {
  tags = append(tags, attachment.Tags...)
}

if blocked, blocker := keeper.IsAnyBlocked(ctx, creator, tags); blocked {
  return sdkerrors(sdkerrors.ErrInvalidRequest, 
      fmt.Sprintf("The user with address %s has blocked you", blocker))
}

return nil
```

Reply:
Unfortunately here we don't need to check which of the tags the post's creator blocked but from which of them he's blocked. So those checks are inevitable for the moment. 
We could open another issue with the aim to reduce the gas costs of create posts txs.

Reply:
You're right. I thought we had to check if the user blocked any of the tags, but actually it's the other way around.